src/build.d

```diff
@@ -1301,23 +1274,23 @@ Params:
 Returns:
     `true` if the target is up to date
 */
-auto isUpToDate(string[] targets, string[][] sources...)
+auto isUpToDate(R, S)(R targets, S sources)
 {
     if (force)
         return false;
-
+    auto oldestTargetTime = SysTime.max;
```

Comment:
Maybe start with the timestamp of `thisBuildScript` to avoid the `chain` + array literal?

Reply:
This function is supposed to be generic, not specific to this particular `build.d`.

Reply:
Okay

Reply:
BTW, we really ought to start checking the hash of the sources, not just the timestamp. Do you think this would be easily doable (of course I'm not talking about doing it with this PR)?

Reply:
"Source hashing" does make the build more robust, but comes at the cost of performance.  @CyberShadow and I had a discussion about doing this in `rund` (https://github.com/dragon-lang/rund/issues/22).  Right now my thought is that source hashing for rund (and probably for `build.d` as well) is that it's not really worth the performance cost and extra build logic.

Note that I'm assuming that source hashing would only help with "false positives" (rebuilding when you don't need to).  If it also fixes "false negatives" (not rebuilding when you do need to) then I would give it more importance.  If you know of "false negative" use cases it would fix then let me know.

I think my comment here would also apply to source hashing in `build.d`: https://github.com/dragon-lang/rund/issues/22#issuecomment-538777847